### PR TITLE
Use uppercase for CMake functions and its arguments

### PR DESCRIPTION
**Pull request by masterleinad**
As described in https://cmake.org/cmake/help/latest/command/function.html function invocation is case-insensitive, but using the all caps name when it is defined with all lowercase characters is really confusing. Also, the function arguments are in uppercase for the remaining `FUNCTION`s defined.

### CODE REVIEW

**Comment by dalg24** [comment]
Add to whitelist

**Comment by crtrott** [comment]
Do we really want function names to be all caps? I like the options and variables being mostly caps and thus they are distinguishable from function calls. Thoughts?

**Comment by masterleinad** [comment]
I am fine with either if it is consistent.

**Comment by crtrott** [comment]
actually you are right a number of functions are already used all caps right now.

File: cmake/kokkos_cxx_std.cmake [hunk 1]
```diff
@@ -43,7 +43,7 @@ FUNCTION(kokkos_set_cxx_standard_feature standard)
       MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
     ENDIF()
   ENDIF()
-ENDFUNCTION(kokkos_set_cxx_standard_feature)
+ENDFUNCTION(KOKKOS_SET_CXX_STANDARD_FEATURE)
```
Comment (dalg24) [inline cmake/kokkos_cxx_std.cmake]:
```suggestion
ENDFUNCTION()
```

File: cmake/kokkos_functions.cmake [hunk 2]
```diff
@@ -28,7 +28,7 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
     SET(${UC_NAME} ${DEFAULT} PARENT_SCOPE)
   ENDIF()
 
-ENDFUNCTION(kokkos_option)
+ENDFUNCTION(KOKKOS_OPTION)
```
Comment (dalg24) [inline cmake/kokkos_functions.cmake]:
```suggestion
ENDFUNCTION()
```

**Comment by dalg24** [comment]
> Do we really want function names to be all caps? I like the options and variables being mostly caps and thus they are distinguishable from function calls. Thoughts?

I am in favor of
* lowercase keywords
* lowercase user-declared macros/functions with uppercase arguments
* lowercase local variables

**Comment by jjwilke** [comment]
We will return to this after 3.0 release. This is really just going to be aesthetics that doesn't break anything.

**Comment by dalg24** [comment]
@masterleinad if you want to cherry-pick b6bbd25 into a separate PR I will merge it

**Comment by dalg24** [comment]
Adding to Milestone Release 3.1 for tracking and closing for now

**Pull request closed without merging**